### PR TITLE
Create functions to determine the url of a call assignment or survey.

### LINKED ISSUE #1680: Error fetching breadcrumbs in Callers tab of standalone Call Assignment
**Please be as thorough as possible, but if you can't answer everything, please just submit what you have! Thank you!**

## Description
When I go to the Callers tab of a standalone Call Assignment, I get an error from fetching the breadcrumbs. The `undefined` campaign ID is not interpreted as `standalone` and instead just sent as `undefined` in the query string. 
bild

## Steps to reproduce
1. Go to a call assignment that does not belong to a project
2. Go to the Callers tab
3. Notice that it does not have any breadcrumbs at the top of the page
4. Look in the network panel and see a `500` status from the request to get the breadcrumbs

### Expected Behaviour
I get breadcrumbs that look the same as they do on the Overview tab of the same call assignment

### Actual Behaviour
I get no breadcrumbs at all.

### PR DESCRIPTION
**Pull request by ziggabyte**
## Description
This PR adds functions to get urls for surveys and call assignments, so they can be either belonging to a project (campaign) or not.

## Changes
- Adds functions to get url for surveys and call assignments, just like for events. 
- use these functions in respective Layouts. 

## Related issues
Resolves #1680

### CODE REVIEW

File: src/features/surveys/utils/getSurveyUrl.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { ZetkinSurvey } from 'utils/types/zetkin';
+
+export default function getSurveyUrl(survey: ZetkinSurvey | null) {
```
Comment (richardolsson) [inline src/features/surveys/utils/getSurveyUrl.ts:3]: This is exactly the type of function where tests would be great. I don't think you need to add tests now, but it's one of those things that when you get proficient enough in writing tests, it's easier (at least for me) to create the function TDD than it is to develop it while clicking around the web app, with the added benefit that you can feel really confident about the logic.

File: src/features/callAssignments/utils/getCallAssignmentUrl.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { ZetkinCallAssignment } from 'utils/types/zetkin';
+
+export default function getCallAssignmentUrl(
```
Comment (richardolsson) [inline src/features/callAssignments/utils/getCallAssignmentUrl.ts:3]: Same goes here (regarding tests) of course.

**Review by richardolsson — APPROVED** [review]
Nice work! This looks good to me. 💯

File: src/features/surveys/utils/getSurveyUrl.ts (hunk 1 above)
Comment (ziggabyte) [reply src/features/surveys/utils/getSurveyUrl.ts:3]: Agree! Making it a postit to stick on my computer - "Start with a test!"

**Pull request merged**